tfx/components/example_gen/utils.py

```diff
@@ -283,9 +315,9 @@ def _retrieve_latest_span_version(
     if result is None:
       raise ValueError('Glob pattern does not match regex pattern')
 
-    span_str = result.group(SPAN_PROPERTY_NAME)
+    span_elems_str = [result.group(name) for name in span_group_names]
```

Comment:
I feel have the span and date branch here might be more readable instead of using the magic `span_group_names ` and line348~360 

```
if is_match_span:
  span_strs = [...]
  span_int =
elif is_match_date:
  span_strs = 
  span_int = 

if span_int > xxx
```

Reply:
refactored to have more clean branching logic.